src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -337,48 +344,44 @@ private void received2Initialized(Task task) {
-                task.getChainDealId(), task.getTaskIndex());
-
-        // In case there is chainReceipt (that could happen after a timeout if the init transaction takes too long
-        // to be mined, no status update is performed
-        if (!optionalPair.isPresent()) {
-            return;
-        }
-
-        String existingChainTaskId = task.getChainTaskId();
-        String chainTaskId = optionalPair.get().getLeft();
-        ChainReceipt chainReceipt = optionalPair.get().getRight();
-
-        if (chainTaskId.isEmpty() || !chainTaskId.equalsIgnoreCase(existingChainTaskId)) {
-            log.error("Initialize failed [existingChainTaskId:{}, returnedChainTaskId:{}]",
-                    existingChainTaskId, chainTaskId);
-            updateTaskStatusAndSave(task, INITIALIZE_FAILED);
-            updateTaskStatusAndSave(task, FAILED);
-            return;
-        }
-
-        initializing2Initialized(task, chainReceipt);
+        blockchainAdapterService
+                .requestInitialize(task.getChainDealId(), task.getTaskIndex())
+                .filter(chainTaskId -> chainTaskId.equalsIgnoreCase(task.getChainTaskId()))
+                .ifPresentOrElse(chainTaskId -> {
+                    log.info("Requested initialize on blockchain " +
+                            "[chainTaskId:{}]", task.getChainTaskId());
+                    updateTaskStatusAndSave(task, INITIALIZING);
+                    //Watch initializing to initialized
+                    updateTaskRunnable(task.getChainTaskId());
+                }, () -> {
+                    log.error("Failed to request initialize on blockchain " +
+                            "[chainTaskId:{}]", task.getChainTaskId());
+                    updateTaskStatusAndSave(task, INITIALIZE_FAILED);
+                    updateTaskStatusAndSave(task, FAILED);
+                });
     }
 
     private void initializing2Initialized(Task task) {
         // TODO: the block where initialization happened can be found
-        initializing2Initialized(task, null);
-    }
-
-    private void initializing2Initialized(Task task, ChainReceipt chainReceipt) {
-        String chainTaskId = task.getChainTaskId();
-        long initializationBlock = chainReceipt != null? chainReceipt.getBlockNumber() : 0;
-        if (initializationBlock == 0){
-            log.warn("Initialization block is empty, using deal block [chainTaskId:{}" +
-                    ", dealBlock{}]", chainTaskId, task.getDealBlockNumber());
-            initializationBlock = task.getDealBlockNumber();
-        }
-        task.setInitializationBlockNumber(initializationBlock);
-        updateTaskStatusAndSave(task, INITIALIZED, chainReceipt);
-        replicatesService.createEmptyReplicateList(chainTaskId);
+        blockchainAdapterService
+                .isInitialized(task.getChainTaskId())
+                .ifPresentOrElse(isSuccess -> {
+                    if (isSuccess.equals(Boolean.TRUE)) {
```

Comment:
Shouldn't we use `if (isSuccess)` instead of `if (isSuccess.equals(Boolean.TRUE))`?